pkg/kubelet/allocation/allocation_manager_test.go

```diff
@@ -443,6 +444,150 @@ func TestIsPodResizeInProgress(t *testing.T) {
 	}
 }
 
+func TestPodResizeCompleteEventGen(t *testing.T) {
+	type ContainerAllocation struct {
+		Name      string
+		Resources v1.ResourceRequirements
+	}
+
+	type PodResourceSummary struct {
+		//TODO: Resources v1.ResourceRequirements, add pod-level resources here once resizing pod-level resources is supported
+		InitContainers []ContainerAllocation `json:"initContainers,omitempty"`
+		Containers     []ContainerAllocation `json:"containers,omitempty"`
+	}
+
+	type testResources struct {
+		cpuReq, cpuLim, memReq, memLim int64
+	}
+
+	type testContainer struct {
+		allocated testResources
+		actuated  *testResources
+	}
+	containerRestartPolicyAlways := v1.ContainerRestartPolicyAlways
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			UID:       "12345",
+			Name:      "test-pod",
+			Namespace: "default",
+		},
+		Spec: v1.PodSpec{
+			Containers:     []v1.Container{{Name: "c0"}, {Name: "c1"}},
+			InitContainers: []v1.Container{{Name: "c2", RestartPolicy: &containerRestartPolicyAlways}, {Name: "c3"}},
```

Comment:
nit: format the PodSpec for readability. Mostly more newlines and indents for each nested object